apps/src/applab/firebaseCounters.js

```diff
@@ -0,0 +1,124 @@
+/* global Applab */
+
+import Firebase from 'firebase';
+import { getDatabase } from './firebaseUtils';
+
+const TABLE_ROW_COUNT_LIMIT = 10;
+
+/**
+ * Map representing the maximum number of writes allowed over each time interval.
+ *
+ * @type {Object.<string, number>} Map from rate limit interval in seconds
+ *     to the maximum number of operations allowed during that interval.
+ */
+const RATE_LIMITS = {
+  15: 300,
+  60: 600
+};
```

Comment:
Yes, these are a client-side version of something that will soon also be enforced by firebase rules.

Passing these constants down through `CDO` , `appOptions`, `config`, etc to access them here sounds like a great idea. Getting the CDO variables into the firebase rules will be trickier / messier, but should be doable by generating some content containing `.bolt` syntax and concatenating it with our (forthcoming) `firebase.bolt` file. @islemaster @Bjvanminnen if there's a cleaner templating approach or potential templating tool you can think of please let me know, otherwise I'll try the generating/concatenating approach first.